methods.py

```diff
@@ -56,6 +56,17 @@ def disable_warnings(self):
         self.Append(CXXFLAGS=["-w"])
 
 
+def force_optimization_on_debug(self):
+    # 'self' is the environment
+    if not self["target"] in ["debug", "release_debug"] or self["optimize"] == "size":
```

Comment:
We do optimize on `release_debug` already, this is only needed for `debug` (i.e. engine contributors custom builds).

And then no need to check for `optimize` as this option isn't used in `debug`.

Reply:
Makes sense, fixed.